packages/mds-audit/api.ts

```diff
@@ -116,14 +119,27 @@ function api(app: express.Express): express.Express {
   })
 
   /**
-   * Audit middleware to load the audit into locals using the audit_trip_id
+   * Audit middleware to load the audit and latest provider event into locals
    */
   app.use(pathsFor('/trips/:audit_trip_id'), async (req: AuditApiTripRequest, res: AuditApiResponse, next) => {
     try {
       const { audit_trip_id } = req.params
-      if (isValidAuditTripId(audit_trip_id)) {
-        res.locals.audit_trip_id = audit_trip_id
-        res.locals.audit = await readAudit(audit_trip_id)
+      if (!isValidAuditTripId(audit_trip_id)) {
+        return next()
+      }
+      res.locals.audit_trip_id = audit_trip_id
+      res.locals.audit = await readAudit(audit_trip_id)
+      let { provider_id, provider_vehicle_id } = req.body
+      if ((!provider_id || !provider_vehicle_id) && res.locals.audit) {
+        provider_id = res.locals.audit.provider_id
+        provider_vehicle_id = res.locals.audit.provider_vehicle_id
+      }
+      const device = await readDeviceByVehicleId(provider_id, provider_vehicle_id)
+      try {
+        const provider_event = await readEvent(device.device_id)
+        res.locals.provider_event = provider_event
+      } catch (err) {
+        await log.warn('Unable to find provider event to match with audit event', JSON.stringify(err))
```

Comment:
I'm pretty sure the logger will unwrap any JSON you provide, so the `JSON.stringify()` might be unnecessary